src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -541,9 +544,137 @@ std::shared_ptr<ngraph::Function> XmlDeserializer::parse_function(
         }
     }
 
+    // Read meta data from legacy representation
+    if (root.child("rt_info").empty()) {
+        // Legacy representation
+        // meta_data - MO meta
+        // quantization_parameters - NNCF quantization section
+        std::unordered_set<std::string> meta_names = {"meta_data", "quantization_parameters"};
+        read_legacy_meta_data(function, meta_names, root);
+    } else {
+        read_meta_data(function, root.child("rt_info"));
+    }
+
     return function;
 }
 
+class MetaDataParser : public ov::Meta {
+public:
+    MetaDataParser(const std::string& name, const pugi::xml_node& meta) : m_name(name) {
+        m_meta.append_copy(meta);
+    }
+
+    operator const ov::AnyMap&() const override {
+        parse();
+        return m_parsed_map;
+    }
+
+    operator ov::AnyMap&() override {
+        parse();
+        return m_parsed_map;
+    }
+
+private:
+    bool has_attr(const pugi::xml_node& node, const std::string& name = "value") const {
+        auto attr = node.attribute(name.c_str());
+        return !attr.empty();
+    }
+
+    ov::Any parse_value(const pugi::xml_node& node) const {
+        if (has_attr(node)) {
+            return XMLParseUtils::GetStrAttr(node, "value");
+        } else if (std::string(node.name()) == "unset" && has_attr(node, "unset_cli_parameters")) {
+            return XMLParseUtils::GetStrAttr(node, "unset_cli_parameters");
+        } else {
+            return parse_node(node);
+        }
+    }
+
+    ov::AnyMap parse_node(const pugi::xml_node& node) const {
+        ov::AnyMap result;
+        const std::string node_name = node.name();
+        for (const auto& data : node.children()) {
+            const std::string data_name = data.name();
+            // WA for legacy POT config
+            if (data_name == "config" && node_name == "quantization_parameters") {
+                // Read legacy pot config
+                std::stringstream stream;
+                data.print(stream);
+                std::string str_config = stream.str();
+                str_config = std::regex_replace(str_config, std::regex("<config>"), "");
+                str_config = std::regex_replace(str_config, std::regex("</config>"), "");
+                str_config = std::regex_replace(str_config, std::regex("\n"), "");
+                str_config = std::regex_replace(str_config, std::regex("( +)"), " ");
```

Comment:
why do we need to parse anything ? I supposed we can fully rely on pugixml parsing

Reply:
Because POT adds incorrect xml format:
```
<config> { [], 'json }
</config>
```